hooks/pre-commit.d/check-hope-doe-approves.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+declare -A unapproved
+
+for source_file in $(find -type f -wholename */main/**/*.scala); do
+  class_name=$(basename "${source_file}" ".scala")
+  source_lines_count=$(cat "${source_file}" | wc --lines)
+  expected_test_location=$(dirname $(echo "${source_file}" | sed "s/main/test/g"))
+
+  for test in "${expected_test_location}/${class_name}*.scala"; do
+    if [[ -e "${test}" ]]; then
+      test_lines_count=$(cat "${test}" | wc --lines)
+      if [[ "${test_lines_count}" -lt "${source_lines_count}" ]]; then
+        unapproved[${source_file}]+="Add more tests"
+      fi
+    else
+      unapproved[${source_file}]+="Create tests"
+    fi
+  done
+done
+
+cat << EOF
+--------------------------------------------------------------------------------
+-----██--██--██--██---████---█████---█████----████---██--██--██████--█████------
+-----██--██--██--██--██--██--██--██--██--██--██--██--██--██--██------██--██-----
+-----██--██--███-██--██--██--██--██--██--██--██--██--██--██--██████--██--██-----
+-----██--██--██-███--██████--█████---█████---██--██---████---██------██--██-----
+-----██████--██--██--██--██--██------██------██--██----██----██------██--██-----
+------████---██--██--██--██--██------██-------████-----██----██████--█████------
+--------------------------------------------------------------------------------
```

Comment:
I've noticed that there is a typo. "Unappoved" instead of "Unapproved". This is unacceptable!

Reply:
Sounds reasonable, I going to fix it in the next iteration! :thinking:

Reply:
@myazinn, Done, thank you for pointing that out.